src/lib.rs

```diff
@@ -65,6 +66,18 @@ impl<T> Window for T
     }
 }
 
+/// Trait for OpenGL specific operations.
+pub trait OpenGLWindow: Window {
+    /// Returns the address of an OpenGL function if it exist, else returns null pointer.
+    fn get_proc_address(&self, proc_name: &str) -> *const libc::c_void;
+
+    /// Returns true if this context is the current context.
+    fn is_current(&self) -> bool;
+
+    /// Make this context current.
+    fn make_current(&self);
```

Comment:
OK. We can use `&self` for now and see how that works out for the other back-ends.